packages/server/utils/AzureDevOpsServerManager.ts

```diff
@@ -280,20 +282,42 @@ class AzureDevOpsServerManager {
     return {error: firstError, workItems: workItemReferences}
   }
 
-  async getUserStories(instanceId: string, queryString: string | null, isWIQL: boolean) {
-    if (isWIQL) {
-      const customQueryString = queryString
-        ? `Select [System.Id], [System.Title], [System.State] From WorkItems Where ${queryString}`
-        : "Select [System.Id], [System.Title], [System.State] From WorkItems Where [System.WorkItemType] = 'User Story' AND [State] <> 'Closed' AND [State] <> 'Removed' order by [Microsoft.VSTS.Common.Priority] asc, [System.CreatedDate] desc"
+  async getUserStories(
+    instanceId: string,
+    queryString: string | null,
+    projectKeyFilters: string[] | null,
+    isWIQL: boolean
+  ) {
+    let customQueryString = ''
+
+    let projectFilter = ''
+    if (projectKeyFilters && projectKeyFilters.length > 0) {
+      let firstLoop = true
+      for (const projectKey of projectKeyFilters) {
+        if (firstLoop) {
+          firstLoop = false
```

Comment:
+1 a `forEach` loop that checks whether the `index === 0` instead of using the `let firstLoop` might be easier to read